src/bentoml_cli/deployment.py

```diff
@@ -0,0 +1,637 @@
+            "push",
+        ],
+    )
+    @click.option(
+        "--deployment-name",
+        type=click.STRING,
+        help="Name of the deployment.",
+    )
+    @click.option(
+        "--bento",
+        type=click.STRING,
+        help="Bento tag or name.",
+    )
+    @click.option(
+        "--context", type=click.STRING, default=None, help="Yatai context name."
+    )
+    @click.option(
+        "--description",
+        type=click.STRING,
+        default=None,
+        help="Description of the deployment.",
+    )
+    @click.option(
+        "--expose-endpoint",
+        type=click.BOOL,
+        default=None,
+        help="Expose endpoint or not.",
+    )
+    @click.option(
+        "--cluster-name", type=click.STRING, default=None, help="Name of the cluster."
+    )
+    @click.option(
+        "--kube-namespace",
+        type=click.STRING,
+        default=None,
+        help="Kubernetes namespace.",
+    )
+    @click.option(
+        "--resource-instance",
+        type=click.STRING,
+        default=None,
+        help="Resource instance.",
+    )
+    @click.option(
+        "--min-replicas", type=click.INT, default=None, help="Minimum replicas."
+    )
+    @click.option(
+        "--max-replicas", type=click.INT, default=None, help="Maximum replicas."
+    )
+    @click.option(
+        "--mode",
+        type=click.Choice("deployment", "function"),
+        default=None,
+        help="Deployment mode.",
+    )
+    @click.option(
+        "--type",
+        type=click.Choice(["stable", "canary"]),
+        default="stable",
+        help="Deployment target type.",
+    )
+    @click.option(
+        "--runner",
+        cls=NargsOptions,
+        multiple=True,
+        callback=parse_runners_callback,
+        default=None,
+        help="Runners.",
+    )
+    @click.option(
+        "-p",
+        "--push",
+        is_flag=True,
+        default=False,
+        help="Push the bento for deployment if it doesn't exist in bento cloud",
+    )
+    @click.option(
+        "-f",
+        "--force",
+        is_flag=True,
+        default=False,
+        help="Forced push to yatai even if it exists in bento cloud",
+    )
+    @click.option(
+        "-t",
+        "--threads",
+        default=10,
+        help="Number of threads to use for upload",
+    )
+    @output_option
+    def create(  # type: ignore
+        deployment_name: str,
+        bento: str,
+        file: str,
+        context: str,
+        description: str,
+        expose_endpoint: bool,
+        cluster_name: str,
+        kube_namespace: str,
+        resource_instance: str,
+        min_replicas: int,
+        max_replicas: int,
+        type: DeploymentTargetType,
+        mode: DeploymentMode,
+        runner: dict[str, t.Any],
+        output: t.Literal["json", "default"],
+        force: bool,
+        threads: int,
+        push: bool,
+    ) -> DeploymentSchema:
+        """Create a deployment."""
+        if file is not None:
```

Comment:
add a _echo here just sayiing that --file is passed, then only context and cluster_name will be get from environment variable